From 4172035f1cc7cf8f4442aa5b88e7141eb0ac802a Mon Sep 17 00:00:00 2001
From: Bryan Cantrill <bryan@joyent.com>
Date: Tue, 30 Aug 2016 21:04:56 +0000
Subject: [PATCH] OS-5630 lx_fix_netstack() ironically leaks netstack reference
 Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Approved by: Joshua M. Clulow <jmc@joyent.com>

---
 usr/src/uts/common/brand/lx/os/lx_brand.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/src/uts/common/brand/lx/os/lx_brand.c b/usr/src/uts/common/brand/lx/os/lx_brand.c
index 08d504f31d..8b2c5d39c4 100644
--- a/usr/src/uts/common/brand/lx/os/lx_brand.c
+++ b/usr/src/uts/common/brand/lx/os/lx_brand.c
@@ -1114,6 +1114,8 @@ lx_fix_netstack()
 	for (i = 0; i < nports; i++)
 		ports[i] = 0;
 	mutex_exit(lock);
+
+	netstack_rele(ns);
 }
 
 void
-- 
2.21.0

